form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowService.kt

```diff
@@ -39,13 +40,20 @@ class FormFlowService(
     }
 
     fun open(formFlowInstanceId: FormFlowInstanceId) {
+        processExpressions(formFlowInstanceId, FormFlowStep::onOpen)
+    }
+
+    fun complete(formFlowInstanceId: FormFlowInstanceId) {
```

Comment:
Shouldn't this get triggered when a step is completed? Now, in order to complete a step, you need to first make a call to FormFlowInstance.complete, and afterwards make a call to FormFlowService.complete... Either a call needs to be made from this complete method to the instance complete method, or we move this logic there after all.

Reply:
Yes, @Klaas-Ritense pointed out the same issue after submitting the PR. Still figuring out the right solution.

Reply:
done